src/activities/content/ContentWebLinkEntity.js

```diff
@@ -0,0 +1,164 @@
+import { Entity } from '../../es6/Entity';
+import { Actions, Classes } from '../../hypermedia-constants';
+import { performSirenAction } from '../../es6/SirenAction';
+
+/**
+ * ContentWebLinkEntity class representation of a d2l content-weblink entity.
+ */
+export class ContentWebLinkEntity extends Entity {
+
+	/**
+	 * @returns {string} Description html of the content-weblink item
+	 */
+	descriptionRichText() {
+		if (!this._entity) {
+			return null;
+		}
+		const subEntity = this._entity.getSubEntitiesByClass(Classes.content.description)[0];
+		if (!subEntity || !subEntity.properties) {
+			return null;
+		}
+		return subEntity.properties.html || '';
+	}
+
+	/**
+	 * @returns {string} Description text of the content-weblink item
+	 */
+	descriptionText() {
+		if (!this._entity) {
+			return null;
+		}
+		const subEntity = this._entity.getSubEntitiesByClass(Classes.content.description)[0];
+		if (!subEntity || !subEntity.properties) {
+			return null;
+		}
+		return subEntity.properties.text || '';
+	}
+
+	/**
+	 * @returns {boolean} external resource value (i.e. open in new tab or not)
+	 */
+	isExternalResource() {
+		if (!this._entity) {
+			return null;
```

Comment:
hmmm you bring up a good point. I'll change it to false. I doubt this condition will be hit when we call this function anyways but not sure if this would hide a potential error in the UI? I haven't setup any UI for the weblinks so can't confirm this yet.